src/util/taproot.rs

```diff
@@ -448,7 +448,7 @@ impl TaprootBuilder {
 
     /// Checks if the builder is a complete tree.
     pub fn is_complete(&self) -> bool {
-        self.branch.len() == 1 && self.branch[0].is_some()
+        self.branch.len() == 1 && self.branch[0].is_some() && !self.has_hidden()
```

Comment:
I don't think we should be updating this function here. Instead, we should be using it in psbt `TapTree::from_inner()` check. Trees can be complete and well-formed even if they contain hidden nodes.

Reply:
Also thought. But than, is the tree considered "complete" having hidden nodes? May be this method should be named "is_finalizable"?

Reply:
> But than, is the tree considered "complete" having hidden nodes?

I think there are two different things here. Builder is called `complete` if there is nothing else to add. In that sense using `complete` makes sense here. I don't mind renaming this to `is_finalizable`. 

`TapTree` is always complete.

Reply:
`is_finalizable` would be ok but this PR uses `is_finalized` which does not make sense to me